### PR TITLE
Allow check in from item in associations view

### PR DESCRIPTION
**Pull request by rdeutz**
Pull Request for Issue #13999.

### Summary of Changes
Added a function to checkin item from the associations view.

### Testing Instructions
* Go to Components > Multilingual Association
* Filter by articles and english language (it can be another component and/or language)
* Open a Multilingual Association from the results
* Don't click close button, just open a new tab and go to Components > Multilingual Association, so the association will we locked and you will see the lock icon

**Before Patch**

If you click on check-in icon, nothing happens.

**After Patch**

The item is checked in.

**Please also test with other items as articles**

### CODE REVIEW

**Comment by rdeutz** [comment]
Unfortunately, I had to change the language file, I will merge this anyway even if we don't have the time to let all translation teams translate the term. 

@brianteeman can you have a look at the sentence I would call english :-)

File: administrator/components/com_associations/controllers/associations.php [hunk 1]
```diff
@@ -68,4 +68,63 @@ public function clean()
...
+		$types     = $extension->get('types');
+
+		if (!array_key_exists($typeName, $types))
+		{
+			return;
+		}
+
+		if (AssociationsHelper::typeSupportsCheckout($extensionName, $typeName) === false)
+		{
+			// How on earth we came to that point, eject internet
```
Comment (brianteeman) [inline administrator/components/com_associations/controllers/associations.php]: lol

File: administrator/language/en-GB/en-GB.com_associations.ini [hunk 2]
```diff
@@ -49,3 +49,4 @@ COM_ASSOCIATIONS_TITLE_EDIT="Multilingual Associations: Edit Associations (%1s &
 COM_ASSOCIATIONS_TITLE_LIST="Multilingual Associations (%1s &gt; %2s)"
 COM_ASSOCIATIONS_TITLE_LIST_SELECT="Multilingual Associations: Select Item Type and Language"
 COM_ASSOCIATIONS_XML_DESCRIPTION="Improved multilingual content management component"
+COM_ASSOCIATIONS_YOU_ARE_NOT_ALLOWED_TO_CHECKIN_THIS_ITEM="You can't checkin this item"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_associations.ini]: check in two words please
https://developer.joomla.org/en-gb-user-interface-text-guidelines/action-or-description.html

Comment (rdeutz) [inline administrator/language/en-GB/en-GB.com_associations.ini]: "check in" and not checkin ?

**Comment by infograf768** [comment]
`check-in` if I do not mistake (it is at least the way it is displayed in the toolbar)

File: administrator/language/en-GB/en-GB.com_associations.ini (hunk 2 above)
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_associations.ini]: yes 
Check in is two words when it's a verb eg I walked up to the counter to check in

**Comment by brianteeman** [comment]
we have a style guide - please us it

**Comment by rdeutz** [comment]
fixing a bug 1 hour; making the comments right several hours ;-)

**Comment by infograf768** [comment]
I do not want to be the bad boy here but I confirm we use `check-in` 44 times in en-GB lang files...

**Comment by rdeutz** [comment]
so it is wrong 44 time :-P

**Comment by brianteeman** [comment]
or maybe it is being used according to the style guide 44 times.

**Comment by infograf768** [comment]
My test:

1. This works fine when an item displayed in the list has been checked out by the same user who is using multilingual associations. In that sense it solves https://github.com/joomla/joomla-cms/issues/13999

If the item is marked as checked-out, one can check it in and then click to open the side by side.

2. Now, when another user (with no check in ability) is with another browser displaying an item which has been checkedout, we do not get the grey lock but only an absence of link:

Example:
screen shot 2017-05-08 at 10 21 29

We can get this instead
screen shot 2017-05-08 at 10 30 36

with  this change:
Add line 129 of /administrator/components/com_associations/views/associations/tmpl/default.php
```
						<?php if (!$canCheckin && $isCheckout) : ?>
							<?php echo JHtml::_('jgrid.checkedout', $i, $item->editor, $item->checked_out_time, 'associations.'); ?>
						<?php endif; ?>
```
just before
```
						<?php if ($canCheckin && $isCheckout) : ?>
							<?php echo JHtml::_('jgrid.checkedout', $i, $item->editor, $item->checked_out_time, 'associations.', $canCheckin); ?>
						<?php endif; ?>
```

**Comment by infograf768** [comment]
_Note: I am preparing a PR to solve also a checked out issue in the side by side view._

**Comment by infograf768** [comment]
See also https://github.com/joomla/joomla-cms/pull/15889

**Comment by stellainformatica** [comment]
I have tested this item :white_check_mark: successfully on 3f301d18709d4473b6f14b15760b7ba18fd1859b<br /><br />Patch test successfully, with JM patch modification.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15873">issues.joomla.org/tracker/joomla-cms/15873</a>.</sub>

**Comment by AlexRed** [comment]
I have tested this item :white_check_mark: successfully on 3f301d18709d4473b6f14b15760b7ba18fd1859b<br /><br />Patch ok for me<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15873">issues.joomla.org/tracker/joomla-cms/15873</a>.</sub>

File: administrator/components/com_associations/controllers/associations.php [hunk 3]
```diff
@@ -68,4 +68,63 @@ public function clean()
...
+	/**
+	 * Method to check in an item from the association item overview.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function checkin()
+	{
+		// Set the redirect so we can just stop processing when we find a  condition we can't process
```
Comment (Quy) [inline administrator/components/com_associations/controllers/associations.php]: Remove extra space `a  condition`.

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by infograf768** [comment]
@franz-wohlkoenig 
We wait for @rdeutz to merge https://github.com/rdeutz/joomla-cms/pull/14

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 8acee881045d5664a25ba10c6804ed9d3486f471<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15873">issues.joomla.org/tracker/joomla-cms/15873</a>.</sub>

**Comment by infograf768** [comment]
@rdeutz
Can now be merged and thanks!

**Comment by infograf768** [comment]
Note: as soon as we are ready to release 3.7.1, we will let it know to TTs about the new string, but I was unable for now to reproduce an error using it.

File: administrator/components/com_associations/controllers/associations.php (hunk 3 above)
Comment (zero-24) [inline administrator/components/com_associations/controllers/associations.php]: fixed with a direct commit here: https://github.com/joomla/joomla-cms/commit/99a518cd0bdc64d09dd50410274a84f20a40d125 Thanks for spotting.

**Pull request merged**